platform-sdk/swirlds-config-impl/src/test/java/com/swirlds/config/impl/ConfigApiListTests.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.config.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.swirlds.common.config.sources.SimpleConfigSource;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.config.api.ConfigurationBuilder;
+import java.util.List;
+import java.util.NoSuchElementException;
+import org.junit.jupiter.api.Test;
+
+class ConfigApiListTests {
+
+    @Test
+    void readListProperty() {
+        // given
+        final Configuration configuration = ConfigurationBuilder.create()
+                .withSource(new SimpleConfigSource().withIntegerValues("testNumbers", List.of(1, 2, 3)))
+                .build();
+
+        // when
+        final List<Integer> values = configuration.getValues("testNumbers", Integer.class);
+
+        // then
+        assertEquals(3, values.size(), "A property that is defined as list should be parsed correctly");
```

Comment:
I didn't know that a `List` property included the semantics that the list be in the same order as specified in the source property file.  (I guess I thought It was more like a "bag" - as opposed to a `Set`.)

Reply:
Well, we (in this case I) did the List extraction ;) But today I would use Set for it...

Reply:
oh i see, so even the "bag" property isn't interesting (it's just a set - no duplicate elements - represented a different way for historical reasons)

Reply:
(but then the test is too precise since it checks the _order_ of the elements ... I guess it doesn't hurt at this time ...)